scripts/dream.py

```diff
@@ -255,8 +317,25 @@ def image_writer(image, seed, upscaled=False):
                         results.append([path, formatted_dream_prompt])
                 last_results.append([path, seed])
 
+                if opt.save_progress is not None:
+                    nonlocal progress_video_type
+                    if progress_video_type == 'v' or progress_video_type == 'vo':
+                            frames_for_video.append(image)
+                            make_video(frames_for_video, os.path.join(current_outdir, f'{prefix}.{seed}.mp4'))
+                            frames_for_video.clear()
+
+                if opt.show_progress is not None:
+                    show_progress(image)
+                    cv2.waitKey(1000)
+
             catch_ctrl_c = infile is None # if running interactively, we catch keyboard interrupts
+
+            step_callback = None
+            if opt.save_progress is not None or opt.show_progress is not None:
```

Comment:
`if opt.save_progress or opt.show_progress`? Or is it possible for these to be non-None False?

Reply:
Yes I believe so. You are able to provide them both without any values and doing so makes them True but the values themselves are still None.